src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -515,11 +545,363 @@ _context_create( fd_exec_instr_test_runner_t *        runner,
+    _add_to_data( &txn_raw_cur_ptr, &program_id_index, sizeof(uchar) );
+
+    // Compact array of account addresses
+    ushort acct_count = (ushort) test_ctx->tx.message.instructions[i].accounts_count;
+    _add_compact_u16( &txn_raw_cur_ptr, acct_count );
+    for( ushort j = 0; j < acct_count; ++j ) {
+      uchar account_index = (uchar) test_ctx->tx.message.instructions[i].accounts[j];
+      _add_to_data( &txn_raw_cur_ptr, &account_index, sizeof(uchar) );
+    }
+
+    // Compact array of 8-bit data
+    pb_bytes_array_t * data = test_ctx->tx.message.instructions[i].data;
+    if( data ) {
+      ushort data_len = (ushort) data->size;
+      _add_compact_u16( &txn_raw_cur_ptr, data_len );
+      _add_to_data( &txn_raw_cur_ptr, data->bytes, data_len );
+    } else {
+      _add_compact_u16( &txn_raw_cur_ptr, 0 );
+    }
+  }
+
+  /* Address table lookups (N/A for legacy transactions) */
+  ushort addr_table_cnt = 0;
+  if( !test_ctx->tx.message.is_legacy ) {
+    /* Compact array of address table lookups (https://solanacookbook.com/guides/versioned-transactions.html#compact-array-of-address-table-lookups) */
+    // NOTE: The diagram is slightly wrong - the account key is a 32 byte pubkey, not a u8
+    addr_table_cnt = (ushort) test_ctx->tx.message.address_table_lookups_count;
+    _add_compact_u16( &txn_raw_cur_ptr, addr_table_cnt );
+    for( ushort i = 0; i < addr_table_cnt; ++i ) {
+      // Account key
+      _add_to_data( &txn_raw_cur_ptr, test_ctx->tx.message.address_table_lookups[i].account_key, sizeof(fd_pubkey_t) );
+
+      // Compact array of writable indexes
+      ushort writable_count = (ushort) test_ctx->tx.message.address_table_lookups[i].writable_indexes_count;
+      _add_compact_u16( &txn_raw_cur_ptr, writable_count );
+      for( ushort j = 0; j < writable_count; ++j ) {
+        uchar writable_index = (uchar) test_ctx->tx.message.address_table_lookups[i].writable_indexes[j];
+        _add_to_data( &txn_raw_cur_ptr, &writable_index, sizeof(uchar) );
+      }
+
+      // Compact array of readonly indexes
+      ushort readonly_count = (ushort) test_ctx->tx.message.address_table_lookups[i].readonly_indexes_count;
+      _add_compact_u16( &txn_raw_cur_ptr, readonly_count );
+      for( ushort j = 0; j < readonly_count; ++j ) {
+        uchar readonly_index = (uchar) test_ctx->tx.message.address_table_lookups[i].readonly_indexes[j];
+        _add_to_data( &txn_raw_cur_ptr, &readonly_index, sizeof(uchar) );
+      }
+    }
+  }
+
+  /* Set up txn descriptor from raw data */
+  fd_txn_t * txn_descriptor = (fd_txn_t *) fd_scratch_alloc( fd_txn_align(), fd_txn_footprint( instr_count, addr_table_cnt ) );
+  ushort txn_raw_sz = (ushort) (txn_raw_cur_ptr - txn_raw_begin);
+  fd_txn_parse( txn_raw_begin, txn_raw_sz, txn_descriptor, NULL );
+
+  /* Set up txn_raw */
+  fd_rawtxn_b_t raw_txn[1] = {{.raw = txn_raw_begin, .txn_sz = txn_raw_sz}};
+
+  /* Run txn preparation phases */
```

Comment:
Make it clear that this should be changed if the runtime transaction setup logic changes